MainWindow: move shortcuts from .cpp to .ui file + change Clone Entry shortcut

c4rlo opened the pull request:
Where possible, move shortcuts from being defined in `MainWindow.cpp` to being defined in `MainWindow.ui`. Previously, there was a seemingly random mix of both practices.

The only shortcuts defined in the `.cpp` file are now ones that can't be defined in the `.ui` file, because they are in some way conditional.

This also reduces the number of compiler warnings of the kind

    warning: arithmetic between different enumeration types ‘Qt::Modifier’ and ‘Qt::Key’ is deprecated [-Wdeprecated-enum-enum-conversion]

with recent GCC versions.

And some minor code cleanups in `MainWindow.cpp`.

Also, change the shortcut for the Clone Entry shortcut from Ctrl+K to Ctrl+D (see discussion below).

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Refactor (significant modification to existing code)

c4rlo committed:
MainWindow: move shortcuts from .cpp to .ui file

The only shortcuts defined in the .cpp file are ones that can't be
defined in the .ui file, because they are in some way conditional.

This also reduces the number of compiler warnings of the kind

    warning: arithmetic between different enumeration types ‘Qt::Modifier’ and ‘Qt::Key’ is deprecated [-Wdeprecated-enum-enum-conversion]

with recent GCC versions.

c4rlo committed:
MainWindow: cosmetic code cleanup

droidmonkey commented:
You need to mark all the shortcuts in .ui as not translatable

c4rlo commented:
> You need to mark all the shortcuts in .ui as not translatable

Ok, done.

c4rlo commented on src/gui/MainWindow.cpp:
```diff
@@ -2190,55 +2190,30 @@ void MainWindow::initActionCollection()
-    ac->setDefaultShortcut(m_ui->actionLockDatabase, Qt::CTRL + Qt::Key_L);
-    ac->setDefaultShortcut(m_ui->actionLockAllDatabases, Qt::CTRL + Qt::SHIFT + Qt::Key_L);
-    ac->setDefaultShortcut(m_ui->actionEntryEdit, Qt::CTRL + Qt::Key_E);
-    ac->setDefaultShortcut(m_ui->actionEntryDelete, Qt::CTRL + Qt::Key_D);
-    ac->setDefaultShortcut(m_ui->actionEntryDelete, Qt::Key_Delete);
```
The second of these lines was overriding the first, so only Delete was actually bound to this action. Now that this shortcut is moved to the .ui file, it's still only Delete that's bound to the action, so no change in behaviour.

I don't know if there was an idea at some point to have Ctrl+D be a valid alternative delete shortcut.

droidmonkey replied:
Ctrl+D should be used for Cloning (ie, duplicating)

c4rlo replied:
Cloning an entry is currently bound to Ctrl+K – see the line just below.

droidmonkey replied:
This is just the default shortcut, you can add any number of shortcuts as alternates. Although I have never used Ctrl + K since it is entirely obtuse. Klone!

c4rlo replied:
You're saying you'd like to add Ctrl+D as an alternate shortcut for cloning? Perhaps that would be best done in a separate PR?

I'm also not sure how that would interact with the shortcut editor – that seems to only support a single shortcut per action.

droidmonkey replied:
https://doc.qt.io/qt-5/qaction.html#setShortcuts

I'd be happy with changing the default for clone to Ctrl + D

c4rlo replied:
Yes, I had noticed that QAction supports multiple shortcuts. But the KeePassXC shortcut editor UI only supports assigning a single shortcut per action. So pre-defining multiple shortcuts for an action may not work well with that, is what I was getting at.

But ok, I can just change the default for cloning from Ctrl+K to Ctrl+D then (without using multiple shortcuts for that action).

droidmonkey replied:
Yah I removed the ability to assign multiple shortcuts since it didn't work well. Maybe it can be added again in the future.

c4rlo committed:
MainWindow: change Clone Entry shortcut to Ctrl+D

Previously, it was Ctrl+K.

As per
https://github.com/keepassxreboot/keepassxc/pull/10903#discussion_r1641296834.
